runtime/runtime.m

```diff
@@ -372,6 +397,88 @@ void xamarin_framework_peer_unlock ()
 	return mono_class_is_subclass_of (cls, mono_get_array_class (), false);
 }
 
+bool
+xamarin_is_class_nsnumber (MonoClass *cls)
+{
+	// COOP: Reading managed data, must be in UNSAFE mode
+	MONO_ASSERT_GC_UNSAFE;
```

Comment:
The assert only does something when using coop (watchOS) and in debug mode (and in that case we want to hit as many asserts as possible), so I think this is fine as it is.